ui/app/src/Utils/Date.elm

```diff
@@ -32,75 +54,143 @@ units =
     ]
 
 
-timeToString : Time.Time -> String
+timeToString : Posix -> String
 timeToString =
-    round >> ISO8601.fromTime >> ISO8601.toString
+    Iso8601.fromTime
 
 
-term : Parser Time.Time
+term : Parser Float
 term =
-    Parser.map2 (*)
-        Parser.float
-        (units
-            |> List.map (\( unit, ms ) -> Parser.succeed ms |. Parser.symbol unit)
-            |> Parser.oneOf
-        )
-        |. Parser.ignore Parser.zeroOrMore ((==) ' ')
-
-
-durationFormat : Time.Time -> Maybe String
-durationFormat time =
-    if time >= 0 then
+    Parser.succeed (*)
+        |= Parser.float
+        |= (units
+                |> List.map (\( unit, ms ) -> Parser.succeed ms |. Parser.symbol unit)
+                |> Parser.oneOf
+           )
+
+
+addDuration : Float -> Posix -> Posix
+addDuration duration time =
+    Time.millisToPosix <|
+        (Time.posixToMillis time + round duration)
+
+
+timeDifference : Posix -> Posix -> Float
+timeDifference startsAt endsAt =
+    toFloat <|
+        (Time.posixToMillis endsAt - Time.posixToMillis startsAt)
+
+
+durationFormat : Float -> Maybe String
+durationFormat duration =
+    if duration >= 0 then
         List.foldl
             (\( unit, ms ) ( result, curr ) ->
                 ( if curr // ms == 0 then
                     result
+
                   else
-                    result ++ toString (curr // ms) ++ unit ++ " "
-                , curr % ms
+                    result ++ String.fromInt (curr // ms) ++ unit ++ " "
+                , modBy ms curr
                 )
             )
-            ( "", round time )
+            ( "", round duration )
             units
             |> Tuple.first
             |> String.trim
             |> Just
+
     else
         Nothing
 
 
-dateFormat : Time.Time -> String
-dateFormat =
-    Date.fromTime >> (Date.Extra.Format.formatUtc config Date.Extra.Format.isoDateFormat)
```

Comment:
What is the reasoning for partially implementing our own date time library?

Reply:
It is pretty small, the only bits that I added are for formatting. I don't want to pull an extra library for this because of our requirements. We used two libraries, now it is only `Iso8601` for parsing/stringifying and the core `Time` modules.